### PR TITLE
changed ->input[] to ->get_input()

### PR DESCRIPTION
**Pull request by JanMalte**
PR for issue https://github.com/mybb/mybb/issues/1526

### CODE REVIEW

**Comment by ATofighi** [comment]
We can’t automatically merge this pull request.

**Comment by JanMalte** [comment]
The PR should now be able to merge.

**Comment by Destroy666x** [comment]
This PR is still not good. You're:
- not using `get_input()` where it would be nice to see
- using `get_input()` where it's not needed, for example what's the point of calling the function 30 times in usercp.php when the correct value is already assigned to another variable? https://github.com/JanMalte/mybb/blob/feature-1526/usercp.php#L53
- adding the `MyBB::INPUT_STRING` to some places while we decided to skip this parameter since it's default, I think.

**Comment by JanMalte** [comment]
Adding ```MyBB::INPUT_STRING``` was only by merging the changes of @ATofighi 

Assigning the casted value to the input array and accessing it directly isn't the way it was supposed to be used. The code wasn't optimized, as this would make the PR bigger then necessary and would mix different aspects together.
After merging this PR, the code could be optimized by assigning the value to a local variable and remove duplicated method calls. But I would strongly suggest to keep this in a separate PR, as it makes merging much easier.

Using ```MyBB::get_input()``` for multi level array access is not very well tested right now, therefor this was skipped for now, but can come later in a new PR, I think.

**Comment by ATofighi** [comment]
but I don't use `MyBB::INPUT_STRING`...
this is my changes: https://github.com/mybb/mybb/commit/cf871e276d93218e1107f51ec8531e5da174506a
do you see any `MyBB::INPUT_STRING` in it?

**Comment by Destroy666x** [comment]
> Assigning the casted value to the input array and accessing it directly isn't the way it was supposed to be used.

You're missing the "because..." part here. I don't see any reason why not, as long as the proper check is done it doesn't matter if the result is assigned to `$mybb->input['action']` or `$randomvariable`. Not saying that the 2nd isn't better, but the 1st is proper too.

> The code wasn't optimized, as this would make the PR bigger then necessary and would mix different aspects together.

But we won't accept a PR that makes the code much less optimised for no reason, that's the point.

**Comment by Sama34** [comment]
I agree there is no need to replace all `$mybb->input[''action]` with `$mybb->get)input('action')` since, as it was already mentioned it was done that way for performance reasons. And while it may not be a big difference for low frequency visited pages it may be for the rest, `usercp.php` in this case for example.

File: admin/modules/config/mod_tools.php [hunk 1]
```diff
@@ -865,7 +865,8 @@ function checkAction(id)
 
 			if($mybb->get_input('threadprefix', MyBB::INPUT_INT) >= 0)
 			{
-				$thread_options['threadprefix'] = $mybb->get_input('threadprefix', MyBB::INPUT_INT);
+				$thread_options['threadprefix'] = $mybb
+                    ->get_input('threadprefix', MyBB::INPUT_INT);
```
Comment (Destroy666x) [inline admin/modules/config/mod_tools.php]: I meant this line when speaking about the syntax error, but I may have noticed wrong and it possibly doesn't throw anything. But It should be reverted regardless - no need to separate it into a new line.

**Comment by dequeues** [comment]
This PR has merge conflicts that need to be resolved.

**Comment by Destroy666x** [comment]
IMO we should just close this, there's been absolutely no progress since we pointed out inconsistencies and suboptimal code 1.5 year ago.

**Comment by euantorano** [comment]
I'm closing this for now due to the above.

**Pull request closed without merging**